lib/AST/DiagnosticEngine.cpp

```diff
@@ -385,6 +385,23 @@ static bool shouldShowAKA(Type type, StringRef typeName) {
   return true;
 }
 
+/// If a type is part of an argument list which includes another, distinct type
+/// with the same string representation, it should be qualified during
+/// formatting.
+static bool shouldQualifyType(Type type, ArrayRef<DiagnosticArgument> Args) {
+  for (auto arg : Args) {
+    if (arg.getKind() == DiagnosticArgumentKind::Type) {
+      auto argType = arg.getAsType();
+      if (argType &&
+          argType->getCanonicalType() != type->getCanonicalType() &&
```

Comment:
Instead of calling getCanonicalType() you can use the isEqual() method, eg `argType->isEqual(type)`

Reply:
good to know, thanks! I saw the == operator was deleted for Type and missed isEqual() somehow